lib/src/bluetooth_device.dart

```diff
@@ -79,7 +83,7 @@ class BluetoothDevice {
         _connectionSubject!.value == false) {
       yield [];
     }
-    if (!_servicesSubject.hasValue || _servicesSubject.value!.isEmpty) {
+    if (!_servicesSubject.hasValue || _servicesSubject.value.isEmpty) {
```

Comment:
The `!` after `value` was added to support both rxdart 0.26.0 and 0.28.0. This is done because another popular Bluetooth library (Flutter blue) only support 0.26.0 and I don't want people to have to choose between the two libraries.

This is true for all the parts where a value is read from a subject.

Reply:
Unfortunately it is a pain to support both rxdart version. One solution to make the linter/compiler happy is to have something like:
````dart
/// Extensions to easily access value and error.
extension TempRxValueStreamExtensions<T> on ValueStream<T> {
  /// Temp until rxdart 0.27 lands
  T? get compatValueOrNull => hasValue ? value : null;
}
````

and use compatValueOrNull (and we could add a compatValueOrThrow too)

An other option is to get rid of rxdart. reimplementing BehaviorSubject using StreamController is a few lines of code and you won't have any issue with this anymore. The fewer dependencies, the better.

Let me know what you would prefer.

Reply:
I fixed using the compat trick. I tested on both rxdart 0.26 and rxdart 0.27.